### PR TITLE
Publish to NPM

### PR DESCRIPTION
**Pull request by nitro-marky**
Adds NPM publishing to the release workflow in order to publish `@digicatapult/components`, also has a test workflow for PR testing

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -90,3 +107,30 @@ jobs:
...
+          registry-url: 'https://registry.npmjs.org'
+          scope: '@digicatapult'
+      - name: Change Folder
+        run: cd packages/components
+      - name: Install packages
+        run: npm ci
+      - name: Publish to npmjs packages
+        run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (mattdean-digicatapult) [inline .github/workflows/release.yml:136]: these need to be done in a single script not three. Each time the `cwd` will be reset

**Comment by nitro-marky** [comment]
There are a couple of files/deletions that crept in. I'll deal with those seperately

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -3,6 +3,23 @@ name: Build and Tests
 on: [push]
```
Comment (jonmattgray) [inline .github/workflows/release.yml:3]: `release` should only run on `branches: ['main']`. The bump to `0.1.1` on this branch just published to `npm` when it should only run after the PR's merged

**Comment by jonmattgray** [comment]
The READMEs at root and in `packages/components` (they're weirdly similar) still refer to `@digicat` rather than `@digicatapult`. In `index.d.ts` the namespace is `@digicat` which could be confusing

**Comment by nitro-marky** [comment]
I've started the process of renaming

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  preconditions:
+    runs-on: ubuntu-latest
+    outputs:
+      repo_name: ${{ steps.repo_ids.outputs.REPO_NAME }}
+      org_name: ${{ steps.repo_ids.outputs.ORG_NAME }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (dblane-digicatapult) [inline .github/workflows/release.yml:17]: Unnecessary checkout step.

File: yarn.lock [hunk 4]
```diff
@@ -907,9 +907,9 @@
     "exec-sh" "^0.3.2"
     "minimist" "^1.2.0"
 
-"@digicat/components@file:/Users/MAddis/Documents/Code/Hydrogen/ui-component-library/packages/components":
+"@digicatapult/components@file:/Users/MAddis/Documents/Code/Hydrogen/ui-component-library/packages/components":
```
Comment (dblane-digicatapult) [inline yarn.lock:910]: Unsure why yarn is using the file syntax here. Feels wrong, please investigate.

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 4 above)
Comment (nitro-marky) [reply yarn.lock:910]: This is what other repos seem to be using.

File: yarn.lock [hunk 5]
```diff
@@ -909,7 +909,7 @@
 
 "@digicat/components@file:/Users/pmichelevicius/workspace/ui-component-library/packages/components":
```
Comment (mattdean-digicatapult) [inline yarn.lock:910]: can we remove this yarn.lock?

Comment (nitro-marky) [reply yarn.lock:910]: Can do, It's also removed in the following pr

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**